engine/resource/src/mount/mount_generic.cpp

```diff
@@ -60,6 +60,7 @@ namespace dmResource
         if (afi && afi->m_FileResourceData)
         {
             fclose(afi->m_FileResourceData);
+            afi->m_FileResourceData = 0;
```

Comment:
UnmountArchiveInternal is called during reboot and here we close `afi->m_FileResourceData`. This function in turn calls dmResourceArchive::Delete(archive) and there we also try to close the same file descriptor.

Reply:
Then I wonder who should own/close it?
I don't think we should have a second case of "if (fd) clos(fd)" if it is never going to be run?

Reply:
Good point. mount_generic.cpp does not own that file descriptor and should not attempt to close it. Updated the PR!